### ISSUE TITLE
Panic at some point

### ISSUE THREAD

**Issue by gvfnix** [kpetremann/salt-exporter#4]
Hi @kpetremann,
Thank you for creating salt-exporter, really great work!

If you have time, could you please take a look at this panic message?

```
panic: interface conversion: interface {} is []interface {}, not string

goroutine 12 [running]:
github.com/kpetremann/salt-exporter/pkg/events.extractStateFromArgs(...)
	/home/runner/work/salt-exporter/salt-exporter/pkg/events/events.go:47
github.com/kpetremann/salt-exporter/pkg/events.(*SaltEvent).ExtractState(0xc0000a2b60?)
	/home/runner/work/salt-exporter/salt-exporter/pkg/events/events.go:61 +0x47d
github.com/kpetremann/salt-exporter/internal/metrics.ExposeMetrics({0x980978, 0xc000106dc0}, 0xc000072240)
	/home/runner/work/salt-exporter/salt-exporter/internal/metrics/metrics.go:66 +0x89d
created by main.main
	/home/runner/work/salt-exporter/salt-exporter/main.go:62 +0x53e
```

Probably, events with incorrect schema should be just ignored.

**Comment by gvfnix** [comment]
It is the latest release, 0.1.2

**Comment by kpetremann** [comment]
hello @gvfnix,

Thanks for the kind words, much appreciated :) and thanks for reporting the issue!

I'll have a look into it as soon as possible.

In the meantime, would it be possible for you to share the following info:
* which version of Salt are you running?
* what features of Salt you use? (reactors, scheduling, proxy-minions etc...)

**Comment by gvfnix** [comment]
I use the latest version, 3005.1. Features include scheduling, orchestration and reactors. I believe that this very issue was related to the reactor I used to alert about failed `state.apply` functions. This reactor was probably written in an incorrect manner, so the function arguments were not strings, but dicts. In fact, I leveraged salt-exporter to get rid of this reactor as it is unstable and too complex.

**Comment by kpetremann** [comment]
ok this is crystal clear, thanks!
I have the fix, I'll test and release it this evening.

**Comment by kpetremann** [comment]
@gvfnix can you test with "go install github.com/kpetremann/salt-exporter@latest"?

I did not do a new release yet, I'd like to confirm with you it fixes your issue (just in case).

**Comment by kpetremann** [comment]
I meant via `go install github.com/kpetremann/salt-exporter@main`

**Comment by gvfnix** [comment]
Looks good now. Thank you!

**Comment by kpetremann** [comment]
awesome, thanks for checking!

The fix has been released in [v0.1.3](https://github.com/kpetremann/salt-exporter/releases/tag/v0.1.3).

Closing the issue, don't hesitate to create an issue if you spot anything else.
